### ISSUE TITLE
Connection directive with query containing arg named "filter" throws type error

### ISSUE THREAD

**Issue by Durisvk** [dotansimha/graphql-code-generator#3318]
**Describe the bug**

While validating query file with `@connection` directive written correctly, library throws a type error: `GraphQLDocumentError: Expected type UserFilterInput, found ["filter"].`

which is incorrect - `@connection` directive should expect `[String]` type instead of `UserFilterInput`

While manually editing the generated file (the generated document gql` ... my query ... `) and firing request - it works correctly so no type error should be thrown.

**To Reproduce**

https://codesandbox.io/s/angry-shockley-2gnms?fontsize=14&hidenavigation=1&theme=dark

1. My GraphQL schema:

```graphql
type Query {
    user(filter: UserFilterInput!, page: Int, perPage: Int): [User]!
}

input UserFilterInput {
    id: ID!
}

type User {
    id: ID!
    username: String!
    email: String!
}
```

2. My GraphQL operations:

```graphql
query user {
    user(filter: { id: 1 }, page: 1, perPage: 2) @connection(key: "PaginatedUsers", filter: ["filter"]) {
        id
        username
        email
    }
}
```

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
documents: document.graphql
generates:
  types.ts:
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-apollo'
      - 'typescript-resolvers'
```

**Expected behavior**
`filter` argument inside `@connection` directive should be of type `[String]` no matter what query it applies to.

**Environment:**

- OS: MacOS Mojave 10.14.6
- `@graphql-codegen/...`: 1.11.0 (as well as 1.0.7 on my localhost)
- NodeJS: v10.16.0

**Additional context**

**Error log:**
```
AggregateError:
        GraphQLDocumentError: Expected type UserFilterInput, found ["filter"].
            at document.graphql:2:93
        at Object.checkValidationErrors (/sandbox/node_modules/@graphql-toolkit/
common/index.cjs.js:306:15)
        at Object.codegen (/sandbox/node_modules/@graphql-codegen/core/index.cjs
.js:92:16)
    AggregateError:
        GraphQLDocumentError: Expected type UserFilterInput, found ["filter"].
            at document.graphql:2:93
        at Object.checkValidationErrors (/sandbox/node_modules/@graphql-toolkit/
common/index.cjs.js:306:15)
        at Object.codegen (/sandbox/node_modules/@graphql-codegen/core/index.cjs
.js:92:16)


Something went wrong
```

`validate-documents.ts` is probably incorrectly identifying arguments of `query` argument vs `@connection` directive argument

**Comment by Durisvk** [comment]
Hey guys, 

## TLDR;

**2 ways to work around this issue:**

1. either rename your argument (`filter`) in a query to something different - that way the validator won't be confused (in a buggy way) to treat the argument of a directive incorrectly (it wouldn't have the same name)

2. or add this line at the top of your schema definition on a server:


_side note:_ the `@connection` directive is a standard way of handling cached pagination by `apollo-client` library but you must do the following to work around the issue mentioned above.

```graphql
directive @connection(key: String, filter: [String]) on FIELD
```


## Few Points
I've figured out this is not an issue of `graphql-code-generator`
The issue is in `validate-documents.ts` file of `@graphql-toolkit` package (I've managed to create a test case in that library which fails on the valid schema definition).

I was also trying to create a similar test case on `graphql-js` library - mainly in their `validate` function that is being called by `@graphql-toolkit` in `validate-documents.ts`, but unsuccessfully - the query validation there seems correct.


## I'm closing this issue
even though it still remains an issue but it's not related to `graphql-code-generator` package but rather some other package used by this library (most probably `@graphql-toolkit`)
